### PR TITLE
Add documentation of all agent configuration options

### PR DESCRIPTION
**Pull request by KN4CK3R**
While setting up my instance I searched for an overview of all available configuration options but could not find any.

Please have a look at the items with `TODO`. I could not find any information for these items. Feel free to extend these.

### CODE REVIEW

File: docs/docs/30-administration/11-agent-config.md [hunk 1]
```diff
@@ -95,20 +95,89 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
+> Default: `linux/amd64`
+
+Restrict builds by platform conditions.
+
+### `WOODPECKER_FILTER`
+> Default: empty
+
+Filter expression to restrict builds by label.
+
+Example: TODO
```
Comment (anbraten) [inline docs/docs/30-administration/11-agent-config.md:148]: I would not add it to the docs atm at all as it is will be changed in the next release anyways, so we don't have to deprecate it.

File: docs/docs/30-administration/11-agent-config.md [hunk 2]
```diff
@@ -95,20 +95,89 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
+
```
Comment (anbraten) [inline docs/docs/30-administration/11-agent-config.md:149]:
```suggestion
```

File: docs/docs/30-administration/11-agent-config.md [hunk 3]
```diff
@@ -95,20 +95,82 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
-    - WOODPECKER_HEALTHCHECK=true
-    - WOODPECKER_KEEPALIVE_TIME=10
-    - WOODPECKER_KEEPALIVE_TIMEOUT=time.Second * 20
-    - WOODPECKER_GRPC_SECURE=""
-    - WOODPECKER_GRPC_VERIFY=true
-```
+### `WOODPECKER_SERVER`
+> Default: `localhost:9000`
+
+Configures the Woodpecker server address.
```
Comment (6543) [inline docs/docs/30-administration/11-agent-config.md:101]:
```suggestion
Configures gRPC address of the server.
```

File: docs/docs/30-administration/11-agent-config.md [hunk 4]
```diff
@@ -95,20 +95,82 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
+> Default: `x-oauth-basic`
+
+The gRPC username.
+
+### `WOODPECKER_AGENT_SECRET`
+> Default: empty
+
+A shared secret used by server and agents to authenticate communication. A secret can be generated by `openssl rand -hex 32`.
+
+### `WOODPECKER_DEBUG`
```
Comment (6543) [inline docs/docs/30-administration/11-agent-config.md:113]: is deprecated, use WOODPECKER_LOG_LEVEL instead

File: docs/docs/30-administration/11-agent-config.md [hunk 5]
```diff
@@ -95,20 +95,82 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
+
+### `WOODPECKER_HOSTNAME`
+> Default: empty
+
+Configures the agent hostname.
+
+### `WOODPECKER_PLATFORM`
+> Default: `linux/amd64`
+
+Restrict builds by platform conditions.
```
Comment (6543) [inline docs/docs/30-administration/11-agent-config.md:141]: will be hidden in later versions and on compile time instead

**Review by 6543 — CHANGES_REQUESTED** [review]

File: docs/docs/30-administration/11-agent-config.md [hunk 6]
```diff
@@ -95,20 +95,82 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
+### `WOODPECKER_AGENT_SECRET`
+> Default: empty
+
+A shared secret used by server and agents to authenticate communication. A secret can be generated by `openssl rand -hex 32`.
+
+### `WOODPECKER_DEBUG`
+> Default: `true`
+
+Enable agent debug mode.
+
```
Comment (anbraten) [inline docs/docs/30-administration/11-agent-config.md:117]:
```suggestion
```

File: docs/docs/30-administration/11-agent-config.md [hunk 7]
```diff
@@ -95,20 +95,82 @@ See [Conditionals Pipeline](/docs/usage/pipeline-syntax#step-when---conditional-
...
+### `WOODPECKER_HOSTNAME`
+> Default: empty
+
+Configures the agent hostname.
+
+### `WOODPECKER_PLATFORM`
+> Default: `linux/amd64`
+
+Restrict builds by platform conditions.
+
```
Comment (anbraten) [inline docs/docs/30-administration/11-agent-config.md:142]:
```suggestion
```

**Review by 6543 — APPROVED** [review]

**Pull request merged**